stable/spinnaker/templates/configmap/halyard-config.yaml

```diff
@@ -38,14 +38,27 @@ data:
     # Docker Registry
     $HAL_COMMAND config provider docker-registry enable
     {{- range $index, $registry := .Values.dockerRegistries }}
-    
-    $HAL_COMMAND config provider docker-registry account add {{ $registry.name }} --address {{ $registry.address }} \
+
+    if $HAL_COMMAND config provider docker-registry account get {{ $registry.name }}; then
+      PROVIDER_COMMAND='edit'
+    else
+      PROVIDER_COMMAND='add'
+    fi
+
+    $HAL_COMMAND config provider docker-registry account $PROVIDER_COMMAND {{ $registry.name }} --address {{ $registry.address }} \
         --repositories {{ range $index, $repository := $registry.repositories }}{{- $repository }},{{- end }}
     {{- end }}
 
     $HAL_COMMAND config provider kubernetes enable
     {{- range $index, $context := .Values.kubeConfig.contexts }}
-    $HAL_COMMAND config provider kubernetes account add {{ $context }} --docker-registries dockerhub \
+
+    if $HAL_COMMAND config provider kubernetes account get {{ $context }}; then
```

Comment:
Couldn't decide whether to make a wrapper function. Figured since we're only doing this twice for now we can punt on that decision. Objectively I'd like halyard's CLI to be a bit more idempotent so it would be a bit more scriptable without having to resort to this argument behavior.

Reply:
Agreed! Would be nice to have Hal handle this.